pkg/yum/repository.go

```diff
@@ -280,39 +287,47 @@ func ParseRepomdXML(body io.ReadCloser) (Repomd, error) {
 }
 
 // Unzips a compressed body response, then parses the contained XML for package information
+// This uses a BufferedReader to peek at the data to figure out what type of compression to use.
+// This also gets wrapped in a LimitedReader to prevent large files from causing an OOM
+//
 // Returns an array of package data
-func ParseCompressedXMLData(body io.Reader) ([]Package, error) {
-	var reader, data io.Reader
+func ParseCompressedXMLData(body io.Reader, maxSize int64) ([]Package, error) {
+	var reader io.Reader
 	var err error
+	result := []Package{}
+
+	bufferedReader := bufio.NewReader(body)
+
+	// peek at the first bytes to determine the type
+	header, err := bufferedReader.Peek(20)
```

Comment:
I originally thought i'd just throw in something large (like 1000), but if the file isn't that big you get an EOF error on the Peek().  An empty primary.xml.gz file as created by createrepo is 139 bytes.  An empty gzip file (with no contents) is 29 bytes, but that can differ based on the filename of the compressed file.   since 20 passes all the test cases, i think its fine?

Reply:
I think 20 is good. You might even get away with 4? But 20 is good.

Reply:
an empty xz file looks to be 60 bytes, but looking at the hex output:
```
$ xxd test.txt.xz 
00000000: fd37 7a58 5a00 0004 e6d6 b446 0200 2101  .7zXZ......F..!.
00000010: 1600 0000 742f e5a3 0100 000a 0000 0000  ....t/..........
00000020: 6423 3933 2d10 881c 0001 1901 a52c 81cc  d#93-........,..
00000030: 1fb6 f37d 0100 0000 0004 595a            ...}......YZ
```
it looks like the first few bytes are sufficient.  An alternative i guess could be to try to peek more, catch the error and then peek less.  But i guess lets just go with this and we can handle any issues that may pop up